src/Client.php

```diff
@@ -264,23 +265,35 @@ public function getIdealIssuers(): IdealIssuers
      * Only completed orders could be refunded.
      *
      * @param string $order_id
+     * @param Properties\Amount|null $amount
      * @return array
      * @throws \GingerPluginSdk\Exceptions\InvalidOrderStatusException
      * @throws \GingerPluginSdk\Exceptions\RefundFailedException
+     * @throws \GingerPluginSdk\Exceptions\RefundAlreadyDoneException
      */
-    public function refundOrder(string $order_id)
+    public function refundOrder(string $order_id, Properties\Amount $amount = null)
     {
         $order = $this->getOrder(id: $order_id);
 
         if ($order->getStatus()->get() != "completed") {
             throw new InvalidOrderStatusException($order->getStatus()->get(), 'completed');
         }
 
-        if (!$order->getCurrentTransaction()->isCaptured()) {
+        if ($order->getCurrentTransaction()->isCapturable() && !$order->getCurrentTransaction()->isCaptured()) {
```

Comment:
To my mind this check work only for capturable orders.
With two divided checks we will have exception when order doesn't require any capturing.